x/gamm/pool-models/balancer/pool_suite_test.go

```diff
@@ -650,6 +654,56 @@ func (suite *KeeperTestSuite) TestCalcJoinPoolShares() {
 			),
 			expectShares: sdk.NewInt(1250000000000 + 609374990000),
 		},
+		{
+			// Compares expected shares and actual shares to make sure the multiplicativeTolerance ratio below is maintained
+			// Expected shares were determined based on increasing the tokensIn amount, the higher the tokens the more the shares
+			// |a - b| / min(a, b) <= allowedErrRatio
+			// 0.667779634 = abs(6_243_749_992_287-3_743_749_992_293)/3_743_749_992_293
+			// 0.667779634 <= 0.0000001 (NOT WITHIN ACCEPTED RATIO)
+			//
+			// where:
+			// a = expected shares,
+			// b = actual shares
+			// allowedErrRatio = 0.0000001
+			// 	Full solution: https://www.wolframalpha.com/input?i=abs%286243749992287-3743749992293%29%2F3743749992293
+			name:       "Error tolerance: Multi token with unequal amounts, with equal weights and 0.01 swap fee",
+			swapFee:    sdk.MustNewDecFromStr("0.01"),
+			poolAssets: oneTrillionEvenPoolAssets,
+			tokensIn: sdk.NewCoins(
+				sdk.NewInt64Coin("uosmo", 50_000),
+				sdk.NewInt64Coin("uatom", 75_000),
+			),
+			expectShares:         sdk.NewInt(2.5e12 + 1350000000000), // ACTUAL: 6_243_749_992_287
```

Comment:
Also, is there a Wolfram calculation for `expectShares`? How was this value selected?

Reply:
I thought about this previously as well, but if i calculate expShares(through our logic)  and insert the right value, it wouldn't cause tolerance err right? So i purposely tried to mess around the numbers

Reply:
So our goal is to find out if it is possible to find test cases with valid inputs / settings that cause us to fall outside of the error tolerance.

That would mean having a valid calculation for `expectShares` , similar to previous test cases that, at the same time, fall outside of the error tolerance.

How can we claim that calculated `shares` fall outside of the error tolerance from `expectShares` when the `expectShares are randomly selected, right?

So, for this to be accepted, I would like to know how `expectShares` are estimated. Preferably, with a Wolfram link similar to previous test cases.

Reply:
hmm i tried lots of scenarios and i wasn't able to generate scenarios to get values that fall outside the error tolerance. In face the larger the `tokensIn` the better the bound became. Maybe we might wanna make the error tolerance higher?  

This is the largest value i tried; 
```
 Desc: one token in - equal weights with zero swap fee
 P_issued = P_supply * ((1 + (A_t / B_t))^W_t - 1)
 4_880_884_817_015_154_699 =  1e20 * (( 1 + (100000000000 / 1e12))^0.5 - 1) 

|a - b| / min(a, b) <= allowedErrRatio
0.000000000000000000204 = abs(4_880_884_817_015_154_699-4_880_884_817_015_154_700)/4_880_884_817_015_154_699

0.000000000000000000204 <= 0.0000001 (WITHIN ACCEPTED RATIO)
```

Reply:
cc @p0mvn

Reply:
That makes sense, thanks for looking deep into this.

If that's the case, let's do the following:
- add any test cases that you've tried and think make sense to add
- make comments consistent with the rest of the test cases
- remove prints and `expectToleranceError bool`

Then, we can close this :+1:  The conclusion is that we could not find a case to fall outside of the tolerance